### DISCUSSION TITLE
Cookies in OpenApi

### DISCUSSION THREAD

**Discussion by Christoph-AK** [poem-web/poem#164] [Q&A]
Hey there, what is the intended way to set cookies with openapi? 
I would love to store a token in a cookie and later retrieve it as parameter in an oai function.
I kinda see a way where I host an extra path with a 'non-openapi'-Poem-Endpoint or maybe a transform for my login endpoint, but that both seems like a lot of extra work and redundant code needed.

**Comment by sunli829** [comment]
You can use the `poem` extractor directly.

```rust
use poem::session::Session;
use poem::session::CookieSession;

#[OpenApi]
impl Api {
    #[oai(path = "/", method = "post")]
    async fn do_something(&self, session: &Session) {
    }
}

let api_service = OpenApiService::new(Api , "test", "1.0");
let app = Route::new().nest("/api", api_service)
    .with(CookieSession::new(CookieConfig::default())); // <<<< use session middleware
```

**Reply by Christoph-AK** [reply]
Thanks, worked like a charm :)
